packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -516,8 +525,15 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   @visibleForTesting
   Duration get cursorBlinkInterval => _kCursorBlinkHalfPeriod;
 
+  int _obscureShowCharTicksPending = 0;
+  int _obscureLatestCharIndex;
+
   void _cursorTick(Timer timer) {
     _showCursor.value = !_showCursor.value;
+    if (_obscureShowCharTicksPending > 0) {
+      _obscureShowCharTicksPending--;
+      setState(() {});
```

Comment:
Please put a comment inside the `{}` explaining what state changed (or, better, put the actual state change there).